### PR TITLE
Pallet credential

### PR DESCRIPTION
**Pull request by amarts**

### CODE REVIEW

**Comment by amarts** [comment]
This is still WIP. Made sure I had some basic things done and it is currently almost the copy of pallet-schema pallet satish wrote.

With this, will add revoke and verify functions to Pallet.

**Comment by smohan-dw** [comment]
looks good

File: pallets/pallet-credential/src/lib.rs [hunk 1]
```diff
@@ -75,59 +74,83 @@ decl_event!(
...
+        <IssuerOfCred<T>>::insert(&id, &owner);
+        <CredByHash<T>>::append(&hash, &id);
+
+        Self::deposit_event(RawEvent::CredentialRegistered(who, id, hash));
+
+        Ok(())
+    }
+
+    #[weight = 100]
+    pub fn revoke_credential(origin, id: CredId, owner: T::AccountId, hash: T::Hash) -> dispatch::DispatchResult {
```
Comment (smohan-dw) [inline pallets/pallet-credential/src/lib.rs:133]: register and revoke got the same signature. how do we differentiate the response?

Comment (amarts) [reply pallets/pallet-credential/src/lib.rs:133]: Was referencing https://github.com/substrate-developer-hub/substrate-verifiable-credentials/blob/master/assets/5.2-finished-code.rs#L42 while doing it.

Will add tests and see how it works.

**Comment by amarts** [comment]
It is building fine, and tests passing. Will work on cord-ui to make sure this is properly working.

**Pull request merged**